assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3819,8 +3834,24 @@ protected String build_instructor_grade_submission_context(VelocityPortlet portl
             state.setAttribute(NEW_ASSIGNMENT_CHECK_ANONYMOUS_GRADING, assignmentService.assignmentUsesAnonymousGrading(a));
 
             boolean allowToGrade = true;
-            if (StringUtils.isNotBlank(a.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT))) {
+            String assignmentAssociateGradebook = a.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT);
+            if (StringUtils.isNotBlank(assignmentAssociateGradebook)) {
                 String gradebookUid = toolManager.getCurrentPlacement().getContext();
+                // If the assignment reference is not equal to the associated gradebook item, then a custom gb item is being used
+                if (!assignmentRef.equals(assignmentAssociateGradebook)) {
+                    try {
+                        org.sakaiproject.grading.api.Assignment gbAssignment = gradingService.getAssignment(gradebookUid,
```

Comment:
Please put a comment about what differences are between Master and 22.x, what lines did we replace. Thanks

Reply:
The logic between master and 22 here should work the same.
On 22.x an `org.sakaiproject.service.gradebook.shared.Assignment` object is retrieved from the `GradebookService` if it's null we don't set the context variables.
For the master an `org.sakaiproject.service.gradebook.shared.Assignment` object is retrieved from the `GradingService`, if an `AssessmentNotFoundException` (yes Assessment) thrown, the variables are not added to the context.

Reply:
But put a comment int the source! :D